server/datastore/mysql/migrations/tables/20240322145615_CreateTableNanoDDMRequests.go

```diff
@@ -0,0 +1,41 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240322145615, Down_20240322145615)
+}
+
+func Up_20240322145615(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE mdm_apple_declarative_requests (
+  id BIGINT NOT NULL AUTO_INCREMENT,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  enrollment_id VARCHAR(255) NOT NULL,
+  -- Should be one of "tokens", "declaration-items", "status", or "declaration/…/…" where the ellipses reference a declaration on the server
+  message_type VARCHAR(255),
+  -- json payload
+  raw_json TEXT,
```

Comment:
Same here for not null. Also heads-up as it's definitely not obvious, `TEXT` has smaller limits than `MEDIUMTEXT`, to keep in mind depending on the size we expect to have to store here.

Reply:
Any reason not to specify this as `JSON` rather than `TEXT`?

Reply:
@gillespi314 What do you mean?

Reply:
Ah good point, I think Sarah means the `JSON` data type that's natively supported by mysql (see e.g. the `app_config_json` table).

Reply:
Does MySQL 5.7 support JSON column types?